src/software/handheld_controller/controller_primitive_generator_test.cpp

```diff
@@ -28,3 +31,67 @@ TEST(ControllerPrimitiveGeneratorTest, test_create_direct_velocity)
               0.5);
     EXPECT_EQ(direct_velocity_primitive->direct_control().dribbler_speed_rpm(), 200);
 }
+
+TEST(ControllerPrimitiveGeneratorTest, test_create_primitive_controller_input)
+{
+    ControllerInput input = ControllerInput();
+    input.setKickButtonPressed(true);
+
+    auto actual_primitive =
+        *ControllerPrimitiveGenerator::createPrimitiveFromControllerInput(
+            input, DynamicParameters->getHandheldControllerConfig());
+
+    double kick_speed = DynamicParameters->getHandheldControllerConfig()
+                            ->KickSpeedMetersPerSecond()
+                            ->value();
+    auto expected_kick_primitive =
+        *createKickPrimitive(Point(), Angle::zero(), kick_speed);
+    EXPECT_TRUE(google::protobuf::util::MessageDifferencer::Equals(
+        expected_kick_primitive, actual_primitive));
+}
+
+TEST(ControllerPrimitiveGeneratorTest, test_create_primitive_controller_input1)
+{
+    ControllerInput input = ControllerInput();
+    input.setChipButtonPressed(true);
+
+    auto actual_primitive =
+        *ControllerPrimitiveGenerator::createPrimitiveFromControllerInput(
+            input, DynamicParameters->getHandheldControllerConfig());
+
+    double chip_distance =
+        DynamicParameters->getHandheldControllerConfig()->ChipDistanceMeters()->value();
+    auto expected_chip_primitive =
+        *createChipPrimitive(Point(0, 0), Angle::zero(), chip_distance);
+    EXPECT_TRUE(google::protobuf::util::MessageDifferencer::Equals(
+        expected_chip_primitive, actual_primitive));
+}
+
+TEST(ControllerPrimitiveGeneratorTest, test_create_primitive_controller_input2)
+{
+    ControllerInput input = ControllerInput();
```

Comment:
can we actually set x/y/angular velocities for this input?

Reply:
I'm not sure, I haven't tried that but I think the function `controller_primitive_generator.cpp` already sets it to some value in `createPrimitiveFromControllerInput`

Reply:
I'm sure you can set the x/y/angular velocities. Let me know if you have questions